### DISCUSSION TITLE
Is it necessary to 'await' the response of Analytics logEvent function?

### DISCUSSION THREAD

**Discussion by cenobyte321** [invertase/react-native-firebase#4572] [Q&A] [answered]
I've seen in the analytics usage documentation (https://rnfirebase.io/analytics/usage) that the sample code to log an event "awaits" the logEvent function:

```
onPress={async () =>
          await analytics().logEvent('basket', {
            id: 3745092,
            item: 'mens grey t-shirt',
            description: ['round neck', 'long sleeved'],
            size: 'L',
        })
}
```

Is it necessary to call the logEvent function synchronously? What are the tradeoffs of not awaiting the result of this function? I want to call this function in the background since I don't want my app's flow to depend on the analytics system.

**Comment by mikehardy** [comment]
Quite a good question - I don't think it matters at all - it doesn't even return anything! (I just checked)
I suppose it is useful in case there is an error so you don't have an uncaught exception / promise rejection

I've never seen those API calls throw in practice though. Maybe worth a careful attempt?

**Reply by cenobyte321** [reply]
Thanks for the reply! Most of the time, we don't perform any other async functions/promises with that function, so there wouldn't be a need to have the async keyword. 

If I'm not mistaken, we can catch the exception if we do something like`logEvent(...).then(()=> { return; }).catch( (e) => { console.error("Could not log event") })`

**Reply by mikehardy** [reply]
Seems like it would work yes, then you're not blocking your current thread by awaiting and exceptions are consumed.

**Reply by CTOverton** [reply, marked as answer]
> ... 4 years later haha

@cenobyte321 since the logEvent doesn't return data you don't even need the `.then` you can simply do

Here is a very minimal way to write it
```ts
logEvent(...).catch(console.error)
```
